### ISSUE TITLE
Getting Not Authorized on ConvoLink.user using latest amplify CLI @aws-amplify/cli@1.6.6

### ISSUE THREAD

**Issue by rohanray** [aws-samples/aws-appsync-chat-starter-react#9]
I had cloned this repo and got it working without any issue a month back. 

Cloned it today in a new environment which had the latest amplify cli installed `@aws-amplify/cli@1.6.6` 

Tried running below query in appsync web consle

```
query GetUser($id: ID!) {
  getUser(id: $id) {
    id
    username
    registered
    userConversations {
      items {
        id
        name
        status
        conversation {
          id
          name
          createdAt
          associated {
            items {
              convoLinkUserId
              user {
                id
                username
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        __typename
      }
      __typename
    }
    __typename
  }
}
```

Response:

```
{
  "data": {
    "getUser": {
      "id": "xxx-a8ab-4a96-b21f-xxx",
      "username": "congOne",
      "registered": true,
      "userConversations": {
        "items": [
          {
            "id": "xxx-4228-4265-b395-xxx",
            "name": "roray (chat)",
            "status": "CREATING",
            "conversation": {
              "id": "xxx-cc84-4138-9887-xxx",
              "name": "direct",
              "createdAt": "2019-05-03T04:35:21.114Z",
              "associated": {
                "items": [
                  {
                    "convoLinkUserId": "xxx-a8ab-4a96-b21f-xxx",
                    "user": {
                      "id": "xxx-a8ab-4a96-b21f-xxx",
                      "username": "congOne",
                      "__typename": "User"
                    },
                    "__typename": "ConvoLink"
                  },
                  {
                    "convoLinkUserId": "xxx-8ca5-4668-8a71-xxx",
                    "user": null,
                    "__typename": "ConvoLink"
                  }
                ],
                "__typename": "ModelConvoLinkConnection"
              },
              "__typename": "Conversation"
            },
            "__typename": "ConvoLink"
          }
        ],
        "__typename": "ModelConvoLinkConnection"
      },
      "__typename": "User"
    }
  },
  "errors": [
    {
      "path": [
        "getUser",
        "userConversations",
        "items",
        0,
        "conversation",
        "associated",
        "items",
        1,
        "user"
      ],
      "data": null,
      "errorType": "Unauthorized",
      "errorInfo": null,
      "locations": [
        {
          "line": 18,
          "column": 15,
          "sourceName": null
        }
      ],
      "message": "Not Authorized to access user on type ConvoLink"
    }
  ]
}
```

The response resolver that got created for `ConvoLink.user` using `@aws-amplify/cli@1.6.6`

```
## No Static Group Authorization Rules **


## No Dynamic Group Authorization Rules **


## [Start] Owner Authorization Checks **
#set( $isOwnerAuthorized = $util.defaultIfNull($isOwnerAuthorized, false) )
## Authorization rule: { allow: owner, ownerField: "owner", identityField: "sub" } **
#set( $allowedOwners0 = $ctx.result.owner )
#set( $identityValue = $util.defaultIfNull($ctx.identity.claims.get("sub"), "___xamznone____") )
#if( $util.isList($allowedOwners0) )
  #foreach( $allowedOwner in $allowedOwners0 )
    #if( $allowedOwner == $identityValue )
      #set( $isOwnerAuthorized = true )
    #end
  #end
#end
#if( $util.isString($allowedOwners0) )
  #if( $allowedOwners0 == $identityValue )
    #set( $isOwnerAuthorized = true )
  #end
#end
## [End] Owner Authorization Checks **


## [Start] Throw if unauthorized **
#if( !($isStaticGroupAuthorized == true || $isDynamicGroupAuthorized == true || $isOwnerAuthorized == true) )
  $util.unauthorized()
#end
## [End] Throw if unauthorized **

$util.toJson($context.result)
```

The above resolver throws `unauthorized`

## Current workaround

For now I have modified the response resolver to below, which avoids `unauthorized` issue:

```
$util.toJson($context.result)
```

**Comment by awsed** [comment]
Thank you for reporting. Looks like with the new improvements to the Amplify CLI to use @auth to protect @connection fields caused a regression. I'm testing a fix here #11

**Comment by awsed** [comment]
Fix merged
